src/templates/course.js

```diff
@@ -0,0 +1,222 @@
+/**
+ *  Copyright 2020 Adobe. All rights reserved.
+ *  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License. You may obtain a copy
+ *  of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under
+ *  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *  OF ANY KIND, either express or implied. See the License for the specific language
+ *  governing permissions and limitations under the License.
+ */
+
+/** @jsx jsx */
+import { useState } from "react"
+import { css, jsx } from "@emotion/react"
+import { graphql, withPrefix } from "gatsby"
+import CourseNav from "../components/CourseNav"
+import DocLayout from "../components/doclayout"
+import PageActions from "../components/PageActions"
+import SiteMenu from "../components/SiteMenu"
+import RenderMdx from "../components/RenderMdx"
+import NextPrev from "../components/NextPrev"
+import Checkmark from "@spectrum-icons/workflow/Checkmark"
+
+import { Flex, View } from "@adobe/react-spectrum"
+import {
+  ActionButtons,
+  Contributors,
+  Link,
+} from "@adobe/parliament-ui-components"
+
+const findSelectedPageNextPrev = (pathname, pages) => {
+  const flat = flattenPages(pages)
+  const selectedPage = flat.find((page) => {
+    return withPrefix(page.path) === pathname
+  })
+
+  return {
+    nextPage: flat[flat.indexOf(selectedPage) + 1],
+    previousPage: flat[flat.indexOf(selectedPage) - 1],
+  }
+}
+
+const flattenPages = (pages) => {
+  if (pages === null) {
+    return []
+  }
+
+  let flat = []
+  const find = (page) => {
+    flat.push(page)
+
+    if (page.pages) {
+      page.pages.forEach(find)
+    }
+  }
+
+  pages.forEach(find)
+
+  flat = flat.flat()
+  return flat.filter(
+    (page, index) => page.path && page.path !== flat[index + 1]?.path
+  )
+}
+
+const useLocalStorage = (dir, key, initialValue) => {
+  const [visited, flipVisited] = useState(() => {
+    let item = window.localStorage.getItem(dir)
```

Comment:
Trying to use `window` outside of of a `useEffect` when we are generating the pages statically will fail.